Fixed netapp_e_lun_mapping options for backwards compatibilitiy

ndswartz opened the pull request:
##### SUMMARY
Readded options lun and target_type which should have been deprecated instead of being removed.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
netapp_e_lun_mapping

##### ANSIBLE VERSION
```
ansible 2.7.0.dev0 (lun_mapping_pub_update faac31d6e6) last updated 2018/08/24 15:20:12 (GMT -500)
  config file = None
  configured module search path = [u'/home/swartzn/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/swartzn/ansible-dev/lib/ansible
  executable location = /home/swartzn/ansible-dev/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```

gundalow commented on lib/ansible/modules/storage/netapp/netapp_e_lun_mapping.py:
```diff
@@ -89,7 +103,9 @@ def __init__(self):
             target=dict(required=False, default=None),
-            volume_name=dict(required=True)))
+            volume_name=dict(required=True, aliases=["volume"]),
+            lun=dict(required=False),
```
Why has this been added?

ndswartz replied:
We simply found volume_name to be unnecessarily verbose and thought adding volume as an alias would help.

gundalow commented on lib/ansible/modules/storage/netapp/netapp_e_lun_mapping.py:
```diff
@@ -37,6 +39,18 @@
+      - volume
+  lun:
+    description:
+      - This option is deprecated and will have no effect on the module's outcome
```
Confused, why has this been added if it's deprecated?

ndswartz replied:
The previous merge removed the two options, lun and target_type, that should have been deprecate instead for backwards capability.  How should I correct my mistake?

gundalow replied:
So if these options don't do anything that seems to break backwards compatability

ndswartz replied:
If there are existing customer plays that contain the two options, the module will still work as expected but we simplified the logic such that the two options are not needed.  The netapp e-series restapi already had the capability to determine those options without customers explicitly stating them.

gundalow replied:
Could the existing customer set these deprecated options to something other than what the restapi would automatically do, ie surprise the user? 

If these options don't do anything then removing them maybe the right decision. A `note:` should be added in the he modules docs and in `porting_guide_2.7.rst`

gundalow commented on lib/ansible/modules/storage/netapp/netapp_e_lun_mapping.py:
```diff
@@ -37,6 +39,18 @@
+    required: no
+  target_type:
+    description:
+      - This option is deprecated and will have no effect on the module's outcome
```
Why has this been added if it's deprecated?

lmprice replied:
@gundalow There was a patch submitted to resolve a different issue a few days ago. (#44666). This patch removed a few parameters from the module that we realized really weren't necessary. Unfortunately, removing those fields would cause existing playbooks utilizing those parameters to break (that's my fault for not remembering that). This patch is adding those back in (as deprecated fields), to maintain existing playbooks. Does that make sense?

gundalow replied:
If these options don't do anything then I think it may be confusing for the end user to leave them in but not used

lmprice commented:
@ndswartz You'll need to squash this back down to a single commit.

swartzn committed:
Fixed netapp_e_lun_mapping options for backwards compatibility.

Readd lun and target_type as deprecated options.

Note: lun and target_type were removed in patch #44666 since they were
no longer needed for the logic in the module.  However, this cause will
cause existing playbooks utilizing these options to break.

swartzn committed:
Add lun specification and target_type verification for netapp_e_lun_mapping

ndswartz commented:
@gundalow After some discussion resulting from your question about surprising the user, we realized that there is a possibility of ambiguity that a host target and a group target could have the same name. Furthermore it is also possible that the customer could specify a host target_type when in reality it is a group target and vice-versa. 

So, we have added functionality to both the target_type and the lun options.  The target_type option will be optionally used to disambiguate mappable objects and lun will give the user the flexibility to specify the lun value.

ndswartz commented:
@gundalow After some discussion resulting from your question about surprising the user, we realized that there is a possibility of ambiguity that a host target and a group target could have the same name. Furthermore it is also possible that the customer could specify a host target_type when in reality it is a group target and vice-versa. We appreciate your thoughtful considerations.

So, we have added functionality to both the target_type and the lun options.  The target_type option will be optionally used to disambiguate mappable objects and lun will give the user the flexibility to specify the lun value.
